change url in api calls to rewardseverywhere.co

AddisonFreeman opened the pull request:
Let me know if this works.

robgpeak commented:
@AddisonFreeman Please get me on skype. I could not get the new extension to work, but I may be doing something wrong.

robgpeak commented:
I got this error message...
<img width="968" alt="Screen Shot 2019-09-24 at 11 42 05 AM" src="https://user-images.githubusercontent.com/28393235/65527296-70426f80-dec0-11e9-8f35-0bd2d80a8c20.png">

robgpeak commented:
Then I saw this when I clicked on Background Errors...

<img width="1234" alt="Screen Shot 2019-09-24 at 11 44 49 AM" src="https://user-images.githubusercontent.com/28393235/65527516-cb746200-dec0-11e9-815f-da24932b5dc5.png">

cgladue commented:
one second i can fix those, i didnt configure yourcharity, foxwoods, totalrewards, doverdowns in DNS for rewardseverywhere i didnt realize you still used them one moment

AddisonFreeman commented:
Maybe I changed too many urls.

Is foxwood.rewardseverywhere.co correct? I understood that complinks.co should be rewardseverywhere.co on all api calls.

Or is it just for main api calls at api.rewardsecerywhere.co and the sub domains should go unchanged?

Addison Freeman
On Sep 24, 2019, 10:45 AM -0500, robgpeak <notifications@github.com>, wrote:
> Then I saw this when I clicked on Background Errors...
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

cgladue commented:
i didnt know all those subdomains were hard coded, but i added them in DNS now @robgpeak can you try again ?

AddisonFreeman commented:
They are not hardcoded but they are still used. Here’s an example from the code:



Addison Freeman
On Sep 24, 2019, 11:16 AM -0500, cgladue <notifications@github.com>, wrote:
> i didnt know all those subdomains were hard coded, but i added them in DNS now @robgpeak can you try again ?
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

AddisonFreeman commented:
That was the old version which is now:



Addison Freeman
On Sep 24, 2019, 12:17 PM -0500, Addison Freeman <afree1423@gmail.com>, wrote:
> They are not hardcoded but they are still used. Here’s an example from the code:
> <Screen Shot 2019-09-24 at 12.14.54 PM.png>
>
>
> Addison Freeman
> On Sep 24, 2019, 11:16 AM -0500, cgladue <notifications@github.com>, wrote:
> > i didnt know all those subdomains were hard coded, but i added them in DNS now @robgpeak can you try again ?
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub, or mute the thread.

AddisonFreeman commented:
I think it didn't upload from email.

Maybe it's taking awhile for DNS to update.

@cgladue 
new:
<img width="624" alt="Screen Shot 2019-09-24 at 12 24 30 PM" src="https://user-images.githubusercontent.com/5950267/65535221-9e2ab280-dec6-11e9-9a7f-58ce662a1d04.png">
old:
<img width="556" alt="Screen Shot 2019-09-24 at 12 14 54 PM" src="https://user-images.githubusercontent.com/5950267/65535224-9f5bdf80-dec6-11e9-94f0-b363fb525f30.png">

AddisonFreeman commented:
<img width="563" alt="Screen Shot 2019-09-24 at 12 28 10 PM" src="https://user-images.githubusercontent.com/5950267/65535332-cca88d80-dec6-11e9-894b-1f526443f2ca.png">
@cgladue

AddisonFreeman commented:
All of those domains are a result of a request to https://shop.rewardseverywhere.co/api/v1/getSubdomains

robgpeak commented:
@AddisonFreeman I am not as familiar with GitHub as you might think.  I really only use it to submit issues.  I just installed what I think is the latest version, but the version number is still 1.7.4, so I am not sure, and here is a quick video of the steps I went through to install the extension.  Please let me know if I am missing something https://youtu.be/XZETjviVh3w

AddisonFreeman commented:
@cgladue We're still getting 404 errors for subdomain api endpoints:

All these endpoints should be valid to my understanding but are 'Not Found'

https://yourcharity.rewardseverywhere.co/api/v1/getUserDetail
https://foxwoods.rewardseverywhere.co/api/v1/getUserDetail
https://doverdowns.rewardseverywhere.co/api/v1/getUserDetail
https://totalrewards.rewardseverywhere.co/api/v1/getUserDetail
https://foxwoods.rewardseverywhere.co/api/v1/getUserDetail

cgladue commented:
ok one moment



Chad Gladue
*DevNull Technology, LLC*


On Wed, Sep 25, 2019 at 9:56 AM Addison Freeman <notifications@github.com>
wrote:

> @cgladue <https://github.com/cgladue> We're still getting 404 errors for
> subdomain api endpoints:
>
> All these endpoints should be valid to my understanding but are 'Not Found'
>
> https://yourcharity.rewardseverywhere.co/api/v1/getUserDetail
> https://foxwoods.rewardseverywhere.co/api/v1/getUserDetail
> https://doverdowns.rewardseverywhere.co/api/v1/getUserDetail
> https://totalrewards.rewardseverywhere.co/api/v1/getUserDetail
> https://foxwoods.rewardseverywhere.co/api/v1/getUserDetail
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/robgpeak/CL-WebExtension/pull/34?email_source=notifications&email_token=AB4MFFIRMNV47YTDSGFFDTLQLNUW7A5CNFSM4I2BJXVKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7R74DY#issuecomment-535035407>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4MFFJZYO2ZU64XN6VLECTQLNUW7ANCNFSM4I2BJXVA>
> .
>

robgpeak commented:
Anything yet @cgladue ?  Let me know when I can test again.

and @AddisonFreeman what is the procedure for customers who have the current extension installed that want to "upgrade" to the new one?  Do they have to uninstall, and then reinstall, or will it be an update to the existing extension?

AddisonFreeman commented:
Once we upload the new version to the chrome web store installed copies will auto update and users won’t notice a difference.
On Sep 25, 2019, 12:55 PM -0500, robgpeak <notifications@github.com>, wrote:
> Anything yet @cgladue ? Let me know when I can test again.
> and @AddisonFreeman what is the procedure for customers who have the current extension installed that want to "upgrade" to the new one? Do they have to uninstall, and then reinstall, or will it be an update to the existing extension?
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

cgladue commented:
@robgpeak @AddisonFreeman yea i was busy today .. i changed it try again ?

AddisonFreeman commented:
That’s working for me now. I updated the version on the testing branch too. Confirm that on your end Rob and upload to the web store or send me the google account credentials and I’ll do it for you.

Addison Freeman
On Sep 25, 2019, 2:51 PM -0500, cgladue <notifications@github.com>, wrote:
> @robgpeak @AddisonFreeman yea i was busy today .. i changed it try again ?
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

robgpeak commented:
Nope.  Still not working.  It starts off like it's working, but then when I
log into xclub, it stops working.  https://youtu.be/fcdWeqU3ltw

>

AddisonFreeman commented:
Hey @cgladue I'm still getting an error for foxwoods.rewardseverywhere.co all subdomains returned by /getSubdomains need to be working because we're checking to see if a user is logged in. 

I can change the code to fail more gracefully but that is what is preventing the extension from working right now.

AddisonFreeman commented:
@robgpeak turns out it's harder to just check for a bad connection. I could do that but it would make for a slower connection. 

So you can see the issue I made the extension not look for the foxwoods domain, it's working fine for me (besides not being able to log into foxwoods, of course). Chad, let me know when that endpoint is fixed.

cgladue commented:
@AddisonFreeman turns out i added .com instead of .co to the IIS bindings, try it now

robgpeak commented:
@AddisonFreeman @cgladue let me know when this is ready to test again.

AddisonFreeman commented:
I just pushed 1.7.9 and everything looks good from my end.

Addison Freeman
On Sep 26, 2019, 8:45 AM -0500, robgpeak <notifications@github.com>, wrote:
> @AddisonFreeman @cgladue let me know when this is ready to test again.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

AddisonFreeman commented:
Hey @cgladue 
is the response for rewardseverywhere.co/api/v1/authenticate supposed to return a 403 when the user enters a wrong password?

That's what I'm getting but want to make sure that is by design on your end.

Thanks.

cgladue commented:
i believe that is correct yes


Chad Gladue
*DevNull Technology, LLC*


On Fri, Sep 27, 2019 at 11:00 AM Addison Freeman <notifications@github.com>
wrote:

> Hey @cgladue <https://github.com/cgladue>
> is the response for rewardseverywhere.co/api/v1/authenticate supposed to
> return a 403 when the user enters a wrong password?
>
> That's what I'm getting but want to make sure that is by design on your
> end.
>
> Thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/robgpeak/CL-WebExtension/pull/34?email_source=notifications&email_token=AB4MFFN5UYSBQLAXSXMSVI3QLYN2HA5CNFSM4I2BJXVKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7ZFWPQ#issuecomment-535976766>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4MFFK666VZSUUEZAAH453QLYN2HANCNFSM4I2BJXVA>
> .
>
